cmd/commandfuncs.go

```diff
@@ -506,6 +506,15 @@ func cmdValidateConfig(fl Flags) (int, error) {
 	validateCmdConfigFlag := fl.String("config")
 	validateCmdAdapterFlag := fl.String("adapter")
 
+	if validateCmdConfigFlag == "" {
+		validateCmdConfigFlag = "Caddyfile"
+	}
+	_, err := os.Stat(validateCmdConfigFlag)
+	if err != nil {
+		return caddy.ExitCodeFailedStartup,
+			fmt.Errorf("file doesn't exist: %v", err)
```

Comment:
We don't know that the file doesn't exist. It could be any number of errors. I would just return the error.

```suggestion
		return caddy.ExitCodeFailedStartup, err
```

Reply:
Agreed. It could be a permission error.